pkg/testing/ess/deployment.go

```diff
@@ -252,19 +252,18 @@ func (c *Client) DeploymentIsReady(ctx context.Context, deploymentID string, tic
 	defer ticker.Stop()
 
 	statusCh := make(chan DeploymentStatus, 1)
-	errCh := make(chan error)
 
 	for {
 		select {
 		case <-ctx.Done():
 			return false, ctx.Err()
 		case <-ticker.C:
-			statusCtx, statusCancel := context.WithTimeout(ctx, tick)
-			defer statusCancel()
 			go func() {
+				statusCtx, statusCancel := context.WithTimeout(ctx, tick)
+				defer statusCancel()
 				status, err := c.DeploymentStatus(statusCtx, deploymentID)
 				if err != nil {
-					errCh <- err
+					// ignore the error (most likely a transient HTTP error)
```

Comment:
The typical question in this case: would it make sense to at least log this error, to have a trace of what is happening?

Reply:
I agree, every error should be logged, so it's easier to debug CI failures.

Reply:
No logger is passed to this level. I changed the implementation to return the errors that occur, so if it failed to get ready and any errors occurred why checking they will be reported and wrote to stdout.